src/executables/util_cs.c

```diff
@@ -4544,18 +4545,89 @@ flashback (UTIL_FUNCTION_ARG * arg)
 int
 memmon (UTIL_FUNCTION_ARG * arg)
 {
+#if defined(CS_MODE)
   UTIL_ARG_MAP *arg_map = arg->arg_map;
-  char *module;
+  const char *database_name;
+  char *module, *stop;
   bool transaction, show_all;
   int tran_count;
+  int module_index = MMON_MODULE_LAST;
 
   module = utility_get_option_string_value (arg_map, MEMMON_MODULE_S, 0);
   transaction = utility_get_option_bool_value (arg_map, MEMMON_TRANSACTION_S);
   tran_count = utility_get_option_int_value (arg_map, MEMMON_TRAN_COUNT_S);
   show_all = utility_get_option_bool_value (arg_map, MEMMON_SHOW_ALL_S);
 
+  database_name = utility_get_option_string_value (arg_map, OPTION_STRING_TABLE, 0);
+  if (database_name == NULL)
+    {
+      goto print_memmon_usage;
+    }
+
+  if ((!transaction) && tran_count)
+    {
+      fprintf (stderr,
+	       msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_INVALID_TRAN_COUNT_OPTION));
+      goto error_exit;
+    }
+
+  if (tran_count < 0)
+    {
+      goto print_memmon_usage;
+    }
+
+  if (show_all && (transaction || module))
+    {
+      fprintf (stderr, msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_EXCLUSIVE_OPTION));
+      goto error_exit;
+    }
+
+  module_index = strtol (module, &stop, 10);
+  if (module_index)
+    {
+      if (module_index < 0 || module_index > MMON_MODULE_LAST)
+	{
+	  fprintf (stderr, msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_MEMMON, MEMMON_MSG_NO_MATCHING_MODULE),
+		   module);
+	  goto error_exit;
+	}
+    }
+  else
+    {
+      /* module == NULL or module with module name */
+      // TODO: convert module name to module index
+    }
+
```

Comment:
Other utilities does check the validity of the databae name.
```
  if (check_database_name (database_name))
    {
      goto error_exit;
    }
```
Is this out of scope for this issue?

Reply:
I think this protection code is somewhat ambiguous whether it belongs to the scope of this PR. However, since feedback has been received, I will include it.